cosmos/job/JobManager.py

```diff
@@ -81,6 +81,13 @@ def run_tasks(self, tasks):
         map(self.submit_task, tasks, commands)
 
     def terminate(self, is_cleanup=False):
+        """Kills all tasks in a workflow.
```

Comment:
I think the cleanup stuff actually does need to only happen for k8 (we don't want to .kill() finished jobs for other DRMs), I think you should probably separate it into a different a drm-specific function called cleanup() which gets called after terminate the atexit code?  for most DRMs it should default to a no-op, but for k8 it's a special case